src/makeInfinityAdder.js

```diff
@@ -4,7 +4,31 @@
  * @return {function}
  */
 function makeInfinityAdder() {
-  // write code here
+  const adder = function (a) {
+    let sum;
+
+    if (a === undefined) {
+      if (sum === undefined) {
+        sum = 0;
+      }
+
+      return sum;
+    }
+
+    sum = a;
```

Comment:
The assignment 'sum = a;' should be 'sum += a;' to accumulate the sum. Also, this line should be within the else block of the 'if (a === undefined)' check to ensure that 'sum' is only updated when a number is passed.